zppy/templates/coupled_global.py

```diff
@@ -112,28 +113,80 @@ def add_trend(
 # -----------------------------------------------------------------------------
 # Function to get ylim
 def get_ylim(standard_range, extreme_values):
-    standard_min = standard_range[0]
-    standard_max = standard_range[1]
-    if extreme_values == []:
-        return [standard_min, standard_max]
-    extreme_min = np.amin(extreme_values)
-    extreme_max = np.amax(extreme_values)
-    if standard_min <= extreme_min:
-        ylim_min = standard_min
+    if len(extreme_values) > 0:
+        has_extreme_values = True
+        extreme_min = np.amin(extreme_values) - 1
+        extreme_max = np.amax(extreme_values) + 1
+    else:
+        has_extreme_values = False
+        extreme_min = None
+        extreme_max = None
+    if len(standard_range) == 2:
+        has_standard_range = True
+        standard_min = standard_range[0]
+        standard_max = standard_range[1]
     else:
+        has_standard_range = False
+        standard_min = None
+        standard_max = None
+    if has_extreme_values and has_standard_range:
+        # Use at least the standard range,
+        # perhaps a wider window to include extremes
+        if standard_min <= extreme_min:
+            ylim_min = standard_min
+        else:
+            ylim_min = extreme_min
+        if standard_max >= extreme_max:
+            ylim_max = standard_max
+        else:
+            ylim_max = extreme_max
+    elif has_extreme_values and not has_standard_range:
         ylim_min = extreme_min
-    if standard_max >= extreme_max:
+        ylim_max = extreme_max
+    elif has_standard_range and not has_extreme_values:
+        ylim_min = standard_min
         ylim_max = standard_max
     else:
-        ylim_max = extreme_max
+        raise ValueError("Not enough range information suppliede")
     return [ylim_min, ylim_max]
 
 
 # -----------------------------------------------------------------------------
 # Plotting functions
 
+# Generic plot function
+def plot_generic(ax, xlim, exps, var_name, rgn):
```

Comment:
This is the function that allows us to plot any variable we want. The original 8 plots had very specific plotting functions.